daemon/cluster/executor/container/adapter.go

```diff
@@ -146,6 +149,54 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 	return nil
 }
 
+// nodeAttachmentsReady validates that NetworkAttachments exist on this node
+// for every network in use by this task. It blocks until the network
+// attachments are ready, or the context times out. If it returns nil, then the
+// node's network attachments are all there.
+func (c *containerAdapter) nodeAttachmentsReady(ctx context.Context) error {
+	// to do this, we're going to get the attachment store and try getting the
+	// IP address for each network. if any network comes back not existing,
+	// we'll wait and try again.
+	attachmentStore := c.backend.GetAttachmentStore()
+	if attachmentStore == nil {
+		return fmt.Errorf("error getting attachment store")
+	}
+
+	// TODO(dperny): essentially, we're long-polling here. this is really
+	// sub-optimal, but a better solution based off signaling channels would
+	// require a more substantial rearchitecture.
+	poll := time.NewTicker(nodeAttachmentReadyInterval)
+	defer poll.Stop()
+	for {
+		// set a flag ready to true. if we try to get a network IP that doesn't
+		// exist yet, we will set this flag to "false"
+		ready := true
+		for name, attachment := range c.container.networksAttachments {
+			// we only need node attachments (IP address) for overlay networks
+			// TODO(dperny): unsure if this will work with other network
+			// drivers, but i also don't think other network drivers use the
+			// node attachment IP address.
+			if attachment.Network.DriverState.Name == "overlay" {
```

Comment:
Is there any reason we can't do all this waiting in libnetwork (possibly even inside the overlay driver itself)?

Reply:
Waiting in libnetwork would not be possible because libnetwork doesn't have any visibility into the attachment store.  It only creates the networks with the addresses it is given.  It does not know whether they are valid or not.